app/src/main/java/in/hackslash/messsy/onboarding/AccountsUtil.java

```diff
@@ -1,39 +1,135 @@
 package in.hackslash.messsy.onboarding;
 
 
-public class AccountsUtil {
+import android.util.Log;
+import android.util.Patterns;
 
-    private onCompleteListener onCompleteListener;
+import androidx.annotation.NonNull;
 
-    public AccountsUtil createAccount(User user, String password){
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
 
-        // TODO Check if a user is logged in and return ALREADY_LOGGED_IN if a logged in user is found
+import java.util.HashMap;
 
-        // TODO Validate user here and return WRONG_INPUT if anything is wrong
+public class AccountsUtil {
 
-        // TODO Check DB if the user exists and return DUPLICATE_USER if user exists
+    private onCompleteListener onCompleteListener;
+
+    public AccountsUtil createAccount(final User user, String password) {
+
+        String email = user.getEmail();
+        final FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        final FirebaseFirestore firestore = FirebaseFirestore.getInstance();
+        //  Check if a user is logged in and return ALREADY_LOGGED_IN if a logged in user is found
+        if (currentUser != null) {
+            onCompleteListener.onComplete(Status.ALREADY_LOGGED_IN, user);
+        }
+        //  Validate user here and return WRONG_INPUT if anything is wrong
+        else if (!Patterns.EMAIL_ADDRESS.matcher(email).matches() || password.length() < 6) {
+            onCompleteListener.onComplete(Status.WRONG_INPUT, user);
+        }
```

Comment:
You have to check if all of the `User`'s field are non-empty here, not just the email and the password. Put a check for `roomno` also.